packaging/tests/tas-restapi/src/test/java/org/alfresco/rest/tags/DeleteTagsTests.java

```diff
@@ -0,0 +1,61 @@
+package org.alfresco.rest.tags;
+
+import org.alfresco.dataprep.CMISUtil;
+import org.alfresco.rest.model.RestTagModel;
+import org.alfresco.utility.model.TestGroup;
+import org.testng.annotations.Test;
+
+import static org.alfresco.utility.report.log.Step.STEP;
+import static org.springframework.http.HttpStatus.FORBIDDEN;
+import static org.springframework.http.HttpStatus.NOT_FOUND;
+import static org.springframework.http.HttpStatus.NO_CONTENT;
+
+public class DeleteTagsTests extends TagsDataPrep
+{
+    /**
+     * Check we can delete a tag by its id.
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteTag()
+    {
+        STEP("Create a tag assigned to a document and send a request to delete it.");
+        document = dataContent.usingUser(adminUserModel).usingSite(siteModel).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+        RestTagModel aTag = createTagForDocument(document);
+
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().usingTag(aTag).deleteTag();
+        restClient.assertStatusCodeIs(NO_CONTENT);
+
+        STEP("Ensure that the tag has been deleted by sending a GET request and receiving 404.");
+        restClient.authenticateUser(dataUser.getAdminUser()).withCoreAPI().getTag(aTag);
+        restClient.assertStatusCodeIs(NOT_FOUND);
+    }
+
+    /**
+     * Attempt to delete a tag as a regular user and receive 403 error
+     */
+    @Test(groups = {TestGroup.REST_API})
+    public void testDeleteTagAsRegularUser_andFail()
+    {
+        STEP("Create a tag assigned to a document and attempt to delete as a non-admin user");
+        document = dataContent.usingUser(dataUser.getAdminUser()).usingSite(siteModel).createContent(CMISUtil.DocumentType.TEXT_PLAIN);
+        RestTagModel aTag = createTagForDocument(document);
+
+        restClient.authenticateUser(userModel).withCoreAPI().usingTag(aTag).deleteTag();
```

Comment:
The big issue here is that the user does not have access to manage the document at all. Can we change this test so that a Site Manager creates the document and tags it and then tries to remove the tag? It should still fail since the user isn't an admin.

Reply:
Does this matter how (and by whom) the tag gets created in case when we try to use `DELETE /tags/{tagId}` endpoint? We could try with all available non-admin user types here to exhaust the possibilities.

Reply:
My thought was that the Site Manager shouldn't have access and everyone else has fewer permissions.

Reply:
I've modified this test to use a SiteManager user and changed the javadoc to mention that any other user would also behave in the same way when trying to delete a tag. Also, I'm creating the document and adding the Tag as a SiteManager